### DISCUSSION TITLE
Is v2ray without tls safe to `man in the middle attacks`?

### DISCUSSION THREAD

**Discussion by yasmoradi** [v2fly/v2ray-core#2221] [Q&A]
Is `vmess` without tls is safe?
There are lots of topics around the world discussing this and the answer is no, it's not safe!
But in that context, `not safe` means that VPS can easily get detected and blocked without tls.

I've got another concern! Is the government capable to do some kind of `man in the middle attack` (`ssl hijacking` for example) in v2ray endpoints without TLS with a valid certificate?

Even for SSL-enabled v2ray endpoints, when I use CDN which uses SSL Termination behind the scene, is the CDN provider capable to decrypt the data between me and the final destination?

If the answer is `no! that's not safe in those cases!` then are there any differences between HSTS preload-enabled services such as Google with other typical SSL-enabled websites?

Thanks in advance

**Comment by jeremyxn** [comment]
你好！你的邮件我已经收到，我会尽快查看，谢谢！
